### PR TITLE
Convert from Cake to Nuke for build

### PR DESCRIPTION
**Pull request by matt-richardson**
As part of our general move from Cake to Nuke

Uses Octopus.Caching as a source, with customisations based on the existing cake file.

I have compared files from build [0.2.335](https://build.octopushq.com/buildConfiguration/OctopusDeploy_LIbraries_OctoVersion_Ci/2043751?buildTab=artifacts) and build [0.2.336-mattr-convert-to-nuke](https://build.octopushq.com/buildConfiguration/OctopusDeploy_LIbraries_OctoVersion_Ci/2043870?buildTab=artifacts):  
 * number of files match
 *  size of files are roughly the same
 * version numbers are set as expected.

### CODE REVIEW

File: build.cmd [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:; set -eo pipefail
```
Comment (matt-richardson) [inline build.cmd:1]: Boilerplate

File: build.ps1 [hunk 2]
```diff
@@ -0,0 +1,69 @@
+[CmdletBinding()]
```
Comment (matt-richardson) [inline build.ps1:1]: Boilerplate

File: build.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
```
Comment (matt-richardson) [inline build.sh:1]: Boilerplate

File: build/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[*.cs]
```
Comment (matt-richardson) [inline build/.editorconfig:1]: Boilerplate

File: build/Configuration.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+using System;
```
Comment (matt-richardson) [inline build/Configuration.cs:1]: Boilerplate

File: build/_build.csproj [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (matt-richardson) [inline build/_build.csproj:1]: Mostly boilerplate

File: build/_build.csproj.DotSettings [hunk 7]
```diff
@@ -0,0 +1,27 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
```
Comment (matt-richardson) [inline build/_build.csproj.DotSettings:1]: Boilerplate

File: build/Build.cs [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    Target Compile => _ => _
+        .DependsOn(Restore)
+        .Executes(() =>
+        {
+            DotNetBuild(s => s
+                .SetProjectFile(Solution)
+                .SetConfiguration(Configuration)
+                .EnableNoRestore()
+                .SetVersion(OctoVersionInfo.FullSemVer)
+                .SetInformationalVersion(OctoVersionInfo.InformationalVersion));
```
Comment (matt-richardson) [inline build/Build.cs:57]: This is a change - we were using FullSemVer here before

**Review by uglybugger — APPROVED** [review]
This looks very tidy :)

File: build/Build.cs [hunk 9]
```diff
@@ -0,0 +1,107 @@
+// ReSharper disable RedundantUsingDirective - prevent PrettyBot from getting confused about unused code.
```
Comment (uglybugger) [inline build/Build.cs:1]: This looks useful - if that's all we need to make NUKE and R# play nicely together then that's a good win.

**Review by Silcoish — APPROVED** [review]
Looks good Matt - Code changes look comparable to cake, output files are in the same format and all are there, version numbers look accurate :+1: Another one down!

File: build/Build.cs (hunk 9 above)
Comment (matt-richardson) [reply build/Build.cs:1]: Yup - this is the only thing that we need to worry about for the prettybot script. This has been running "in the real world" in OctopusCLI and appears to be working well :success:

**Pull request merged**